lighthouse-core/lib/network-recorder.js

```diff
@@ -341,10 +341,16 @@ class NetworkRecorder extends EventEmitter {
     const records = networkRecorder.getRecords().filter(record => record.isValid);
 
     // create a map of all the records by URL to link up initiator
-    const recordsByURL = new Map();
+    const possiblePrefetchRecords = new Map();
+    const typedRecords = new Map();
     for (const record of records) {
-      if (recordsByURL.has(record.url)) continue;
-      recordsByURL.set(record.url, record);
+      if (record.type === NetworkRequest.TYPES.Other) {
```

Comment:
would love some comments here now that we're getting fancier and catching issues.

maybe 
```
// The initiator is unlikely to be a prefetch record which have type 'Other'.
// Create two maps of initiators, one for prefetch, one for regular, that contain the earliest request for the URL.
```